Assets/Plugins/Source/Core/Utility/FileUtility.cs

```diff
@@ -33,26 +33,80 @@ namespace PlayEveryWare.EpicOnlineServices.Utility
     public static class FileUtility
     {
         /// <summary>
-        /// Generates a unique and new temporary directory inside the Temporary Cache Path as determined by Unity,
-        /// and returns the fully-qualified path to the newly created directory.
+        /// Generates a unique and new temporary directory inside the Temporary
+        /// Cache Path as determined by Unity, and returns the fully-qualified
+        /// path to the newly created directory. The directory name will be
+        /// "Temp-" appended with a GUID.
         /// </summary>
-        /// <returns>Fully-qualified file path to the newly generated directory.</returns>
-        public static string GenerateTempDirectory()
+        /// <param name="path">
+        /// Fully-qualified path to the newly generated directory if one was
+        /// created, otherwise path is set to default System.String value.
+        /// </param>
+        /// <returns>
+        /// True if the temporary directory was created, false otherwise.
+        /// </returns>
+        public static bool TryGetTempDirectory(out string path)
         {
+            path = default;
+            
+            // Nested local function to reduce repetitive code.
+            string GenerateTempPath()
+            {
+                return Path.Combine(
+                    Application.temporaryCachePath,
```

Comment:
For reasons relating to particular platform's reaction to asking for things like Application.temporaryCachePath, can you make a static function or something that this value can be grabbed from? That'd allow easy extension on other platforms for getting this. Otherwise I guess those changes can just be made in-line here.

Reply:
I'm not certain. That is worth considering, but is outside the scope of these changes.